app/Jobs/UpdateWikiSiteStatsJob.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Wiki;
+use App\WikiSiteStats;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+
+class UpdateWikiSiteStatsJob extends Job implements ShouldBeUnique
+{
+    public $timeout = 3600;
+    public function handle (): void
+    {
+        $allWikis = Wiki::all();
+        foreach ($allWikis as $wiki) {
+            $this->updateSiteStats($wiki);
+        }
+    }
+
+    private function updateSiteStats (Wiki $wiki): void
+    {
+        $response = Http::withHeaders([
+            'host' => $wiki->getAttribute('domain')
+        ])->get(
+            getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=query&meta=siteinfo&siprop=statistics&format=json'
+        );
+
+        if ($response->failed()) {
+            $this->job->markAsFailed();
+            Log::error(
+                'Failure polling wiki '.$wiki->getAttribute('domain').' for sitestats: '.$response->clientError()
+            );
+            return;
+        }
+
+        $responseBody = $response->json();
+        $update = [];
+        foreach (WikiSiteStats::FIELDS as $field) {
+            $value = data_get($responseBody, 'query.statistics.'.$field, null);
```

Comment:
I cannot recall what we discussed in the breakdown session: this means if a key is missing, we'll use stale data instead of 0. If we want 0, we can still change it.

Reply:
Could you elaborate on what you mean exactly? Would you change this behavior?

Reply:
For the record, we can't `null` the value itself as it's a numeric non-nullable field.

Reply:
sorry I for the misunderstanding, I meant `stale data`. I don't know why I said `null`

Reply:
Ok, so this is currently using stale data already. Do we need to change anything?

Reply:
nope, I prefer using stale date instead of 0 because 0 can cause confusion